ipywidgets_bokeh/kernel.py

```diff
@@ -39,10 +48,13 @@ def send(self, stream, msg_type, content=None, parent=None, ident=None, buffers=
         msg = self.msg(msg_type, content=content, parent=parent, header=header, metadata=metadata)
         msg['channel'] = stream.channel
 
-        from bokeh.document.events import MessageSentEvent
-
         doc = self.document
-        doc.on_message("ipywidgets_bokeh", self.receive)
+
+        # Ensure document message handler is only added once
+        try:
+            doc.remove_on_message("ipywidgets_bokeh", self.receive)
+        finally:
+            doc.on_message("ipywidgets_bokeh", self.receive)
```

Comment:
Don't love this of course but previously we were attaching this callback multiple times.

Reply:
Was `on_message` not idempotent before bokeh 2.4?